### PR TITLE
Fix various button paddings

### PR DESCRIPTION
**Pull request by SammysHP**
Still not perfect, but looks better than before. The whole area should
be refactored as the extends and overrides are a bit ugly.

### CODE REVIEW

**Comment by megax** [comment]
@SammysHP Not a bad idea, but we shouldn't work with the fractions of pixels. Considering the current situations the whole buttonsystem should be revised. We shouldn't unite it just now as it is, we should wait for a more complete version. It only screws with the code if we do it now. ( but the idea is great :) )

**Comment by khurshid-alam** [comment]
@SammysHP With fixed padding, image button becomes rectangular from square 
image-button

**Comment by SammysHP** [comment]
@khurshid-alam Hmm, probably because I removed `min-height` and `min-width` because this added too much padding to buttons with "normal sized" content (e.g. single line text buttons). The icons are not square and thus the buttons are not square as well.

We could add `min-*` again but that would heavily depend on the icon and/or font size. Button code is a big mess and I fully agree with @megax:

> Considering the current situations the whole buttonsystem should be revised.

PS: Independent of the way we do this, we should try to keep the lines on the left section of the first page of gtk3-widget-factory aligned.

PPS: How do you get these icons? On my system it looks like this:



edit: I forgot that the icon theme not only changes the file type icons, but also everything else. I use Numix icons which have this ugly find icon.

File: gtk-3.20/scss/widgets/_button.scss [hunk 1]
```diff
@@ -45,7 +45,7 @@
     }
 
     %button {
-        padding: $spacing ($spacing + 2px);
+        padding: $spacing ($spacing * 1.4);
```
Comment (satya164) [inline gtk-3.20/scss/widgets/_button.scss:48]: Even though it works, it shouldn't be unitless

File: gtk-3.20/scss/widgets/_toolbar.scss [hunk 2]
```diff
@@ -33,11 +33,11 @@
     button.linked, .linked button { @include linked_button($bg); }
 
     combobox, button {
-        padding: $spacing - 1px;
+        padding: $spacing - 1px; // FIXME Why override everything, must be applied again for .text-button
```
Comment (satya164) [inline gtk-3.20/scss/widgets/_toolbar.scss:36]: what's this FIXME?

File: gtk-3.20/scss/widgets/_view.scss [hunk 3]
```diff
@@ -59,7 +59,7 @@
     }
 
     %column_header_button {
-        padding: 0 6px;
+        padding: ($spacing * 0.5) $spacing ($spacing * 0.2) $spacing;
```
Comment (satya164) [inline gtk-3.20/scss/widgets/_view.scss:62]: Even though it works, it shouldn't be unitless

File: gtk-3.20/scss/widgets/_toolbar.scss [hunk 4]
```diff
@@ -33,11 +33,11 @@
...
 
-        &.text-button { padding: $spacing; }
+        &.text-button { padding: $spacing ($spacing * 1.4); }
```
Comment (satya164) [inline gtk-3.20/scss/widgets/_toolbar.scss:38]: Even though it works, it shouldn't be unitless

File: gtk-3.20/scss/widgets/_toolbar.scss [hunk 5]
```diff
@@ -33,11 +33,11 @@
...
 
-        &.image-button { padding: ($spacing + 1px) ($spacing - 1px) ($spacing + 1px) $spacing; }
+        &.image-button { padding: $spacing; } // FIXME Side effect with toolbar, order important (toolbar buttons might have both, text-button and image-button)
```
Comment (satya164) [inline gtk-3.20/scss/widgets/_toolbar.scss:40]: Even though it works, it shouldn't be unitless

File: gtk-3.20/scss/widgets/_button.scss (hunk 1 above)
Comment (SammysHP) [inline gtk-3.20/scss/widgets/_button.scss:48]: It has a unit – in the `$spacing` variable (`5px * 1.4 = 7px`). The same for your other comments.

File: gtk-3.20/scss/widgets/_toolbar.scss (hunk 2 above)
Comment (SammysHP) [inline gtk-3.20/scss/widgets/_toolbar.scss:36]: This seems to be obsolete and I forgot to remove it.

**Comment by khurshid-alam** [comment]
@SammysHP This has to wait. I don't think your padding variation changes in proportional way. You are basically trying to reduce button padding without keeping it proportional. For example, lets reduce `$spacing` to `3px`, then everything is messed up which shouldn't happen (and doesn't happen normally) for at least  within the range of ($spacing +/- 2).

Although I agree that padding is far for perfect at the moment. Its big and messy because we can't control gtk-icon-size which is either 16x16 or 24x24. What we can do is to reduce it little bit among all widgets and then try your idea (fraction of pixel).

**Comment by SammysHP** [comment]
After many great contributions of @megax this is now obsolete in some parts. The remaining issue is that buttons are too large IMHO. But it requires more effort than in this PR to fix it properly.

**Pull request closed without merging**